src/nvim/ex_session.c

```diff
@@ -690,18 +690,16 @@ static int makeopens(FILE *fd, char_u *dirnow)
       return FAIL;
     }
 
-    //
-    // Save current window layout.
-    //
-    PUTLINE_FAIL("set splitbelow splitright");
-    if (ses_win_rec(fd, tab_topframe) == FAIL) {
-      return FAIL;
-    }
-    if (!p_sb && put_line(fd, "set nosplitbelow") == FAIL) {
-      return FAIL;
-    }
-    if (!p_spr && put_line(fd, "set nosplitright") == FAIL) {
-      return FAIL;
+    if (tab_topframe->fr_layout != FR_LEAF) {
+      // Save current window layout.
+      PUTLINE_FAIL("let s:save_splitbelow = &splitbelow");
```

Comment:
I thought of combining multiple `PUTLINE_FAIL` into a single `fprintf` because Neovim's session files always use LF for line endings instead of using CRLF on Windows.

@justinmk left comments on `put_line` and `put_eol` that they are not needed.

Reply:
Ah, I just had a quick look at https://github.com/neovim/neovim/commit/2c1d12d0b and I see what you mean.